execution/utility.sh

```diff
@@ -1138,14 +1138,23 @@ function get_cloudformation_stack_output() {
   local stackName="$1"; shift
   local resourceId="$1"; shift
   local attributeType="$1"; shift
+  local mode="$1"; shift
+
+  if [[ -z "${mode}" ]]; then
+    mode="enable"
+  fi
 
   if [[ -z "${attributeType}" || "${attributeType}" == "ref" ]]; then
     stackOutputKey="${resourceId}"
   else
     stackOutputKey="${resourceId}X${attributeType}"
   fi
 
-  stack_id="$(aws --region "${region}" cloudformation list-stacks --stack-status-filter "CREATE_COMPLETE" "UPDATE_COMPLETE" --query "StackSummaries[?StackName == '$stackName'].StackId" --output text || return $?)"
+  if [[ "${mode}" -eq "disable" ]]; then
+    stack_id="$(aws --region "${region}" cloudformation list-stacks --stack-status-filter "CREATE_COMPLETE" "UPDATE_COMPLETE" "UPDATE_IN_PROGRESS" "DELETE_IN_PROGRESS" "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS" --query "StackSummaries[?StackName == '$stackName'].StackId" --output text || return $?)"
+  else
+    stack_id="$(aws --region "${region}" cloudformation list-stacks --stack-status-filter "CREATE_COMPLETE" "UPDATE_COMPLETE" --query "StackSummaries[?StackName == '$stackName'].StackId" --output text || return $?)"
+  fi
```

Comment:
What's the intention with this? We shouldn't have stacks in progress when this is called? 
I would have thought they would be deleted stacks.But why would we want the output from something in a deleted stack?

Reply:
I found that the stacks were still in the process of being deleted when this was occurring, because the WAF creation was within the stack I assumed, hence we needed to accept stacks in the state of "UPDATE_IN_PROGRESS" "DELETE_IN_PROGRESS" "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS". 

It may be possible to remove the "CREATE_COMPLETE" "UPDATE_COMPLETE" but I wasn't certain

Reply:
How did that happen though? The epilogue and prologue scripts are run either before the cfn stack is applied or after the stack has been completed 

So the IN_PROGRESS states shouldn't be visible during most operations of this command?

Reply:
I freely admit I don't understand enough about it (though I'd assumed that the individual events were where the prologue and epilogue fit in), but below is the event trace for updating a LB to set WAF/Enabled:false where you can see the WAF deletion completing before the stack update completes (and it's the starting of the deletion that we care about for this is it not?)
![image](https://user-images.githubusercontent.com/26639131/126252225-5654914b-00e2-4791-830e-d19f21b7bcf0.png)

Reply:
So that is the status of each resource in the stack the template processing works based on the state of the stack instead. So epilogue and prologue only run outside of update-stack or create-stack operations which then wait for the stack to be in a *_COMPLETE or *_FAILED state

Reply:
On every test I ran: stop; stopstart; update (with Enabled:false from true). The latter is the cause of the addition of the "UPDATE_COMPLETE_CLEANUP_IN_PROGRESS" - I didn't have it initially and found it was necessary while testing

I was only looking at the WAF removal, but once the wafACLLogicalId is gathered, do we care if the stack outputs are no longer available? What am I missing?

Reply:
The check you are doing is against the stack status these are the resource statues 

I.e. this one 

<img width="905" alt="image" src="https://user-images.githubusercontent.com/16523764/127127560-e8b820d9-de24-4c66-b29f-d75d19500f75.png">

instead of these ones 

<img width="1152" alt="Screen Shot 2021-07-27 at 19 05 23" src="https://user-images.githubusercontent.com/16523764/127127711-e1d1624d-3f11-46db-821a-b3ea33c40808.png">


When that script is run the Stack state can only be in a COMPLETE or a FAILED state. It can't be in an IN_PROGRESS check 

So essentially we just need to skip running the waf check commands when deleting the load balancer. So I think this is more a check on the STACK_OPERATION global environment variable that is missing instead of looking at the resources.

Reply:
Done

I got caught between disabling WAF and disabling WAF logging. I think both are handled now

Reply:
These operations can probably be removed now as the states listed won't ever be used. Can we clean up the status check

Reply:
Those extra status are required on the check if you disable WAF logging (as distinct from disabling WAF, ie what I did in error but I believe is still a valid change)